assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -633,21 +633,30 @@ angular.module('openshiftConsole')
     };
 
     $scope.updateTriggers = function() {
+
+      // helper for creating a webhook object
+      function createWebhook(type) {
+        var webhook = {
+          type: type
+        };
+        webhook[(type === "GitHub") ? "github" : "generic"] = {
+          secret: ApplicationGenerator._generateSecret()
+        };
+        return webhook;
+      }
       var presentTriggers = $scope.triggers.present;
       var triggers = [];
+
       if (presentTriggers.webhook) {
-        var webhooks = _.filter($scope.buildConfig.spec.triggers, function(obj) { return obj.type === "GitHub" })
+        var webhooks = _.filter($scope.buildConfig.spec.triggers, function(obj) { return obj.type === "GitHub" || obj.type === "Generic"})
         if (webhooks.length === 0) {
-          webhooks.push({
-            github: {
-              secret: ApplicationGenerator._generateSecret()
-            },
-            type: "GitHub"
-          });
+          webhooks.push(createWebhook("GitHub"));
+          webhooks.push(createWebhook("Generic"));
+        } else if (webhooks.length === 1) {
```

Comment:
Well thats true, but I remember that while working on BC editor we decided to go just with one checkbox. The point was (if I remember correctly) that it doesn't make sense to create just Generic webhook.
But I see your point, so if anything I can update the logic to distinguish between those two webhooks, since now we dont.